### PR TITLE
feat: update friends list with new design

### LINKED ISSUE #39: Update friends list
### **Objective:**

We want to update the friend list component so that it can match we the updated design 

### **Description:**
 - Update the ```Friend``` card from the ```components```  folder to contain the avatar and the "user info card".
 - Update the title `Who can see`, and use the new updated typography component 
 - Each new list item now should have a checkbox , name, username and avatar profile picture with new updated fonts 
 - The checkbox and plus have a new increase in size
 - Below the component we must have a ```SAVE``` button that will allow you to save the changes you made
 - Follow the spacing styles as designed in the Figma
 - Render the component in the editSignal screen

### **Acceptance Criteria:**

- The component is updated with new design 
- The component displays the title `Who can see` and the friendsList
- When an item is checked in the friends' list, it should have the same styles and changes as is in the Figma 
- The component follows the styles in the Figma
- The component is rendered in the EditSignal screen

### **Design:**
[Figma](https://www.figma.com/design/ZcPlin9PqvfVKaeM8VfUov/SocialApp?node-id=1454-2578&node-type=frame&t=BqAet377l0I9eLzY-11)
Image

### PR DESCRIPTION
**Pull request by serapieTuyishime**
Closes #39

### CODE REVIEW

File: components/lists/Friends.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+import { ScrollView, StyleSheet, View } from "react-native"
 import CustomText from "@/components/ui/CustomText"
 import FriendCard from "@/components/Friend"
-import { visibleFriends } from "@/data/friends"
 import { useState } from "react"
+import { defaultUsers } from "@/data/users"
 
 export default function FriendsList() {
-  const [friendsList, setfriendsList] = useState(visibleFriends)
+  const [friendsList, setfriendsList] = useState(defaultUsers)
```
Comment (Odile078) [inline components/lists/Friends.tsx:8]: let's rename `setfriendsList` to `setFriendsList`

File: screens/EditSignal/index.tsx [hunk 2]
```diff
@@ -20,34 +24,68 @@ export default function EditSignal() {
...
+        </CustomText>
         <CrossMark onPress={() => navigation.push("Home")} />
-        <CustomButton variant="primary" title="Done" textSize="sm" />
       </View>
-      <UserAvatar
-        imageUrl={require("@/assets/images/user-avatar.png")}
-        size="large"
+      <ScrollView
+        contentContainerStyle={{
+          gap: 20,
```
Comment (Jonath-z) [inline screens/EditSignal/index.tsx:34]: I think you might need to configure the spacing as well, maybe not now.

**Review by Odile078 — APPROVED** [review]

**Review by Jonath-z — APPROVED** [review]

File: components/lists/Friends.tsx [hunk 3]
```diff
@@ -1,44 +1,44 @@
...
-          renderItem={({ item }) => (
-            <FriendCard
-              handleChange={() => updateFriendsList(item.id)}
-              user={item}
-            />
-          )}
-          keyExtractor={(item) => item.id.toString()}
-        />
-      </SafeAreaView>
+      <ScrollView
```
Comment (AnasAlamin) [inline components/lists/Friends.tsx:21]: why did you change this to a  `ScrollView`? we can't have 2 scrollviews with the same orientation under each other.

**Review by AnasAlamin — CHANGES_REQUESTED** [review]
please check the comment below

**Review by AnasAlamin — APPROVED** [review]

**Pull request merged**